example/misc/infinite_set.cpp

```diff
@@ -49,6 +50,11 @@ constexpr auto doubleton(X x, Y y) {
 
 namespace boost { namespace hana {
     template <>
+    struct SetTheoretic<infinite_set_tag> {
+        static constexpr bool value = true;
+    };
```

Comment:
If the point of the `SetTheoretic` concept is to guarantee that all associated functions are implemented, then being able to just opt in to use `union` in this case is probably incorrect.

For this example, I would suggest either implementing the remaining functions or just make a custom `union` function. (not sure which is preferable)

Reply:
The usual way of doing things in Hana is that if you satisfy the minimal complete definition, then you automatically model the concept, no need to specialize it. I think this is what we should do here. The only concept that requires specialization is `Sequence`, and that's because there's no additional syntax added on top of existing concepts, only semantics. But this is not the case here, so I'd like the model to be deduced automatically.

Reply:
Makes sense. Removing it.